wallet/src/Pos/Wallet/Web/Pending/Worker.hs

```diff
@@ -136,12 +136,16 @@ processPtxsOnSlot
     => SendActions m -> SlotId -> m ()
 processPtxsOnSlot sendActions curSlot = do
     ptxs <- getPendingTxs
-    let sortedPtxs =
-            sortWith _ptxCreationSlot $
-            flip fromMaybe =<< topsortTxs wHash $
-            ptxs
-
-    processPtxs sendActions curSlot sortedPtxs
+    -- We topsort the transactions (so that we wouldn't try to send a
+    -- transaction before its dependencies), and then we choose one random
+    -- tx without dependencies and send it.
+    let mbComps = topsortTxs wHash ptxs
+    let chooseRand = liftIO . fmap head . shuffleM
+    randomRoot <- case mbComps of
+        Nothing    -> chooseRand ptxs
+        Just comps -> (head =<<) <$> chooseRand comps
```

Comment:
Can you explain why we needed a topsort that returned the components to be able to do this? Why not just pick the first element of the topsort?